sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -942,538 +1338,418 @@ matrix([["createPoller", "LroEngine"]] as const, async function (implName: Imple
                       [headerName]: pollingPath,
+                      "retry-after": "0",
                     },
-                    body: `{"properties":{"provisioningState":"Accepted"},"id":"100","name":"foo"}`,
+                    body: `{"status":"Accepted"}`,
                   },
                   {
                     method: "GET",
                     path: pollingPath,
                     status: 202,
                     headers: {
-                      location: pollingPath,
+                      location: locationPath,
                       [headerName]: pollingPath,
+                      "retry-after": "0",
                     },
                     body: `{"status":"Accepted"}`,
                   },
                   {
                     method: "GET",
                     path: pollingPath,
                     status: 200,
-                    headers: {
-                      location: pollingPath,
-                      [headerName]: pollingPath,
-                    },
-                    body: `{"status":"Canceled"}`,
+                    body: `{"status":"Succeeded", "resourceLocation": "${locationPath}"}`,
+                  },
+                  {
+                    method: "GET",
+                    path: locationPath,
+                    status: 200,
+                    body: `{"id":"100","name":"foo"}`,
                   },
                 ],
-              }),
-              {
-                messagePattern: /Operation was canceled/,
-              }
-            );
+              });
+              assert.deepInclude(result, { id: "100", name: "foo" });
+            });
+
+            it("should handle postAsyncRetrycanceled", async () => {
+              const pollingPath = "/postasync/retry/canceled/operationResults/200/";
+              await assertError(
+                runLro({
+                  routes: [
+                    {
+                      method: "POST",
+                      status: 202,
+                      headers: {
+                        location: "/postasync/retry/succeeded/operationResults/foo/200/",
+                        [headerName]: pollingPath,
+                        "retry-after": "0",
+                      },
+                      body: `{ "properties": { "provisioningState": "Accepted"}, "id": "100", "name": "foo"}`,
+                    },
+                    {
+                      method: "GET",
+                      path: pollingPath,
+                      status: 200,
+                      body: `{"status":"Canceled"}`,
+                    },
+                  ],
+                }),
+                {
+                  messagePattern: /Operation was canceled/,
+                }
+              );
+            });
+          });
+        }
+      );
+
+      describe("LRO Sad scenarios", () => {
+        it("should handle PutNonRetry400 ", async () => {
+          await assertDivergentBehavior({
```

Comment:
The diff in this file is crazy but the main change is using `assertDivergentBehavior` to perform the LRO when `throwOnNon2xxResponse` is true and when it is false.